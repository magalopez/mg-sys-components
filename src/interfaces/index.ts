import { Props as ProductButtonsProps } from '../components/ProductCard/ProductButtons'
import { Props as ProductCardProps } from '../components/ProductCard/ProductCard';
import { Props as ProductImageProps } from '../components/ProductCard/ProductImage'
import { Props as ProductTitleProps } from '../components/ProductCard/ProductTitle'

export interface Product {
  id    : string;
  img?  : string;
  title : string;
}

export interface ProductContextProps {
  counter    : number;
  maxCount?  : number;
  product    : Product;
  increaseBy : (value: number) => void;
}

export interface ProductCardHOCProps {
  ({ children, product, className } : ProductCardProps) : JSX.Element,
  Buttons : (Props : ProductButtonsProps) => JSX.Element;
  Image   : (Props : ProductImageProps) => JSX.Element;
  Title   : (Props : ProductTitleProps) => JSX.Element;
}

export interface OnChangeArgs {
  count   : number;
  product : Product;
}

export interface ProductInCart extends Product {
  count : number
}

export interface ShoppingCartProps {
  [key: string] : ProductInCart
}

export interface InitialValuesProps {
  count?    : number;
  maxCount? : number;
}

export interface ProductCardHandlersArgs {
  count             : number;
  isMaxCountReached : boolean;
  maxCount?         : number;
  product           : Product;

  handleDecrement?  : () => void;
  handleIncrement?  : () => void;
  increaseBy        : (value: number) => void;
  reset             : () => void;
}