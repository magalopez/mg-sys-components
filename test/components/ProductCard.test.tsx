import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components'; // Asegúrate de importar tu componente

const { act } = renderer;
describe('ProductCard', () => {
  test('should show ProductCard', () => {
    const product = {
      id: '1',
      title: 'ProductCard Hellow',
      img: 'https://product-card-image-hellow.jpg'
    };

    const component = renderer.create(
      <ProductCard product={product}>
        {() => (
          <>
            <ProductCard.Image />
            <ProductCard.Title />
            <ProductCard.Buttons />
          </>
        )}
      </ProductCard>
    );

    console.log(component.toJSON());
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should increase product quantity', () => {
    const product = {
      id: '1',
      title: 'ProductCard increase',
      img: 'https://product-card-image-hellow.jpg'
    };

    const component = renderer.create(
      <ProductCard product={product}>
        {({ count, increaseBy }) => (
          <>
            <h1>Increase</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(2)}>Increase</button>
          </>
        )}
      </ProductCard>
    );

    let tree = component.toJSON();
    
    act(() => {
      (tree as any).children[2].props.onClick();
    });

    tree = component.toJSON(); // Actualiza el árbol después del clic

    console.log('Counter: ', (tree as any).children[1].children);
    expect((tree as any).children[1].children[0]).toBe('2');
  });
});