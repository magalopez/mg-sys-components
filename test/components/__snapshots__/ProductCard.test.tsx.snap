// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should show ProductCard 1`] = `
<div
  className="productCard "
>
  <img
    alt="Product Image"
    className="productImg "
    src="https://product-card-image-hellow.jpg"
  />
  <span
    className="productDescription "
  >
    ProductCard Hellow
  </span>
  <div
    className="buttonsContainer "
  >
    <button
      className="buttonMinus"
      onClick={[Function]}
    >
      -
    </button>
    <div
      className="countLabel"
    >
      0
    </div>
    <button
      className="buttonAdd false"
      disabled={false}
      onClick={[Function]}
    >
      +
    </button>
  </div>
</div>
`;
